doc/reference/misc.md

```diff
@@ -1292,6 +1292,75 @@ limit: 0/3
 ```
 :::
 
+## Read/Write Locks
+
+The `:std/misc/rwlock` include an implementation of read write locks.
+
+The implementation preempts readers when there are writers waiting, so
```

Comment:
Wait shouldn't it be the contrary?

My understanding is that read lock holders don't want the locked stuff to change while they're doing stuff, so a writer can't preempt but must be queued.

Thus, when requesting a lock:
- If there are no existing lock holders, get the lock
- If there are existing read lock holders and you are also reading and there is no queue of writers, get the lock
- Otherwise get queued

And when releasing, atomically:
- If you were the last reader or a writer and there is write queue, pop and activate the first queued writer
- If you were a writer and there is no queued writer, but queued readers, pop and activate all readers.

Reply:
What it does is the right thing, otherwise writers will be starved. It is the same brhavior as go's read/write locks.

Basically when trying to acquire read lock, succeed unless there is an exclusive write lock or there are writers waiting.

Reply:
Sorry the wording in the docs is wrong, there is no preemption.

Reply:
Basically implementation does exactly what you described

Reply:
fixed the documentation; the word preempt was just wrong.